app/models/vm_retire_request.rb

```diff
@@ -0,0 +1,17 @@
+class VmRetireRequest < MiqRequest
+  TASK_DESCRIPTION  = 'VM Retire'.freeze
+  SOURCE_CLASS_NAME = 'Vm'.freeze
+  ACTIVE_STATES     = %w(retired) + base_class::ACTIVE_STATES
+
+  validates :request_state, :inclusion => { :in => %w(pending finished) + ACTIVE_STATES }, :message => "should be pending, #{ACTIVE_STATES.join(", ")} or finished"
```

Comment:
This line breaks the UI change in the [associated classic-ui PR](https://github.com/ManageIQ/manageiq-ui-classic/pull/3409)